project.clj

```diff
@@ -7,6 +7,26 @@
     :password :env/nexus_jenkins_password
     :sign-releases false })
 
+(def heap-size-from-profile-clj
+  (let [profile-clj (io/file (System/getenv "HOME") ".lein" "profiles.clj")]
+    (if (.exists profile-clj)
+      (-> profile-clj
+        slurp
+        read-string
+        (get-in [:user :puppetserver-heap-size])))))
+
+(defn heap-size
+  [default-heap-size heap-size-type]
+  (or
+    (System/getenv "PUPPETSERVER_HEAP_SIZE")
+    heap-size-from-profile-clj
+    (do
+      (println "Using" default-heap-size heap-size-type
```

Comment:
Hmm, yeah, that's what I tried.  And still `lein version` displays the messages for me.  It seems like just having it in place at all - even if the profile isn't being used - is enough to have the messages displayed.  I see them if I do:

```clj
:profiles {:blah {
                    :jvm-opts [
                           ~(str "-Xms" (heap-size "1G" "min"))
                           ~(str "-Xmx" (heap-size "2G" "max"))]}
```

... and just run `lein version`.  Seems like the code is resolved even if for a profile that you don't have in use?